src/cli-sdk/src/print-err.ts

```diff
@@ -51,52 +73,40 @@ export const printErr = (
     return lines.join('\n')
   }
 
-  // This is an error with a cause, check if it we know about its
-  // code and try to print it. If it did not print then fallback
-  // to the next option.
-  if (isErrorWithCode(err) && print(err, usage, stderr, format)) {
-    return
+  const err = parseError(e)
+  const knownError = printCode(err, usage, stderr, format)
+  const fileWritten =
+    !knownError || knownError.file ? writeErrorLog(e, format) : null
+
+  // We could not write an error log and its not a know error,
+  // so we print the entire formatted value.
+  if (!fileWritten && !knownError) {
+    return stderr(format(e))
   }
 
-  // We have a real but we dont know anything special about its
-  // properties. Just print the standard error properties as best we can.
-  if (err instanceof Error) {
+  if (err && !knownError) {
     stderr(`${err.name}: ${err.message}`)
-    if ('cause' in err) {
-      stderr(`Cause:`)
-      if (err.cause instanceof Error) {
-        stderr(indent(format(err.cause)))
-      } else if (err.cause && typeof err.cause === 'object') {
-        for (const key in err.cause) {
-          stderr(
-            indent(
-              `${key}: ${format((err.cause as Record<string, unknown>)[key])}`,
-            ),
-          )
-        }
-      } else {
-        stderr(indent(format(err.cause)))
-      }
-    }
-    const stack = trimStack(err)
-    if (stack) {
-      stderr(`Stack:`)
-      stderr(indent(format(stack)))
-    }
-    return
   }
-
-  // We don't know what this is, just print it.
-  stderr(`Unknown Error:`, format(err))
+  if (fileWritten) {
+    stderr('')
+    stderr(`Full details written to: ${fileWritten}`)
+  }
+  if (!knownError || knownError.bug) {
+    stderr('')
+    stderr('Open an issue with the full error details at:')
+    stderr(indent('https://github.com/vltpkg/vltpkg/issues/new'))
```

Comment:
It'd be good to make this an ANSI hyperlink to the URL that has a title, label, and details already injected into the body, if we can.

https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/creating-an-issue#creating-an-issue-from-a-url-query

Reply:
If the contents are too big to put on the query string, we could just have a section that says "Paste your error log file here" or something.

Reply:
Oh! another possible trick, if we're on macOS, we can pipe the details into the `pbcopy` program, and then tell them to paste into an issue.